File: Purifier.java
import java.util.regex.Pattern;

public class Purifier {
    public static final Pattern p = Pattern.compile("[\\W\\d]+", Pattern.UNICODE_CHARACTER_CLASS);

    public static String[] purify(final String text) {
        return p.splitAsStream(text.toLowerCase())
                .filter(w -> w.length() > 3)
                .toArray(String[]::new);
    }
}

File: src/Purifier.scala
/**
  * Created by the-magical-llamicorn on 3/30/17.
  */
object Purifier {

  val pattern = "[\\P{L}\\p{N}]+".r

  def purify(text: String): Seq[String] = {
    pattern.split(text)
  }
}
